### PR TITLE
Get owned namespaces

### PR DESCRIPTION
**Pull request by NemTechCN**

### CODE REVIEW

**Comment by dgarcia360** [comment]
Good job @NemTechCN! It will be great to have the possibility to pass the address to check as an option.

**Comment by dgarcia360** [comment]
@NemTechCN We can get rid of the "optional" option picking the profile address by default, but overwriting it if the address option is set. 

See how we are doing it with ``nem2-cli account info``:

https://github.com/nemtech/nem2-cli/blob/master/src/commands/account/info.ts#L53

**Comment by NemTechCN** [comment]
Thank you for your advice.

File: src/commands/namespace/owns.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                    'Introduce the address: '));
+        } catch (err) {
+            console.log(options);
+            throw new ExpectedError('Introduce a valid address');
+        }
+        const namespaceHttp = new NamespaceHttp(profile.url);
+        const namespaceService = new NamespaceService(namespaceHttp);
+        namespaceHttp.getNamespacesFromAccount(address)
+            .subscribe((namespacesInfo) => {
+                this.spinner.stop(true);
```
Comment (dgarcia360) [inline src/commands/namespace/owns.ts]: You have to start the spinner in line 60 first.

File: src/commands/namespace/owns.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                if (namespacesInfo.length === 0) {
+                    console.log('No namespace exists with profile Address ' + address.plain());
+                } else {
+                    let text = '';
+                    namespacesInfo.map(namespaceInfo => {
+                        namespaceService.namespace(namespaceInfo.id).subscribe(namespace => {
+                            text += chalk.green('Namespace: ') + '\n';
+                            text += '-'.repeat('Namespace: '.length + namespacesInfo.length) + '\n';
```
Comment (dgarcia360) [inline src/commands/namespace/owns.ts]: It should be  ('Namespace: '.length + **namespacesInfo.id.fullName.length)**

File: src/commands/namespace/owns.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        const namespaceService = new NamespaceService(namespaceHttp);
+        namespaceHttp.getNamespacesFromAccount(address)
+            .subscribe((namespacesInfo) => {
+                this.spinner.stop(true);
+                if (namespacesInfo.length === 0) {
+                    console.log('No namespace exists with profile Address ' + address.plain());
+                } else {
+                    let text = '';
+                    namespacesInfo.map(namespaceInfo => {
+                        namespaceService.namespace(namespaceInfo.id).subscribe(namespace => {
```
Comment (dgarcia360) [inline src/commands/namespace/owns.ts]: I don't think you have to call the namespaceService. Try to fetch the namespace name doing:

``namespaceInfo.id.fullName``

File: src/commands/namespace/owns.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                            text += chalk.green('Namespace: ') + '\n';
+                            text += '-'.repeat('Namespace: '.length + namespacesInfo.length) + '\n';
+                            text += 'Name:\t' + namespace.name + '\n';
+                            text += 'ID:\t' + namespace.id.toHex().toUpperCase() + '\n';
+                            text += 'Owner:\t\t' + namespaceInfo.owner.address.pretty() + '\n';
+                            text += 'StartHeight:\t' + namespaceInfo.startHeight.compact() + '\n';
+                            text += 'EndHeight:\t' + namespaceInfo.endHeight.compact() + '\n';
+                            text += 'HasAlias:\t' + namespaceInfo.hasAlias() + '\n';
```
Comment (dgarcia360) [inline src/commands/namespace/owns.ts]: Consider printing the alias. The alias could be an address or a mosaic id, depending on its type.

Comment (dgarcia360) [inline src/commands/namespace/owns.ts]: Output as well if it's a subnamespace or not:

https://github.com/nemtech/nem2-cli/blob/master/src/commands/namespace/info.ts#L95

File: src/commands/namespace/owns.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                    console.log('No namespace exists with profile Address ' + address.plain());
+                } else {
+                    let text = '';
+                    namespacesInfo.map(namespaceInfo => {
+                        namespaceService.namespace(namespaceInfo.id).subscribe(namespace => {
+                            text += chalk.green('Namespace: ') + '\n';
+                            text += '-'.repeat('Namespace: '.length + namespacesInfo.length) + '\n';
+                            text += 'Name:\t' + namespace.name + '\n';
+                            text += 'ID:\t' + namespace.id.toHex().toUpperCase() + '\n';
+                            text += 'Owner:\t\t' + namespaceInfo.owner.address.pretty() + '\n';
```
Comment (dgarcia360) [inline src/commands/namespace/owns.ts]: Try to follow the same format (camelCase) and naming conventions as https://github.com/nemtech/nem2-cli/blob/master/src/commands/namespace/info.ts#L95

File: src/commands/namespace/owns.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
+/*
```
Comment (dgarcia360) [inline src/commands/namespace/owns.ts]: Rename the file owns.ts to owned.ts

File: src/commands/namespace/owns.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            console.log(options);
+            throw new ExpectedError('Introduce a valid address');
+        }
+        const namespaceHttp = new NamespaceHttp(profile.url);
+        const namespaceService = new NamespaceService(namespaceHttp);
+        namespaceHttp.getNamespacesFromAccount(address)
+            .subscribe((namespacesInfo) => {
+                this.spinner.stop(true);
+                if (namespacesInfo.length === 0) {
+                    console.log('No namespace exists with profile Address ' + address.plain());
```
Comment (dgarcia360) [inline src/commands/namespace/owns.ts]: 'The address' + address.plain() + 'does not own any namespaces';

File: src/commands/namespace/owns.ts (hunk 4 above)
Comment (NemTechCN) [reply src/commands/namespace/owns.ts]: namespaceInfo.id.fullName possibly undefined

**Comment by dgarcia360** [comment]
Closed in favor of https://github.com/nemtech/nem2-cli/pull/45

**Comment by dgarcia360** [comment]
btw @NemTechCN, thanks for your PR!

**Pull request closed without merging**